lib/trino-collect/src/main/java/io/trino/collect/cache/NonLoadableCacheImpl.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.collect.cache;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.ForwardingCache;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+
+import static java.util.Objects.requireNonNull;
+
+// package-private. The interface provides deprecation and javadoc to help at call sites
+class NonLoadableCacheImpl<K, V>
+        extends ForwardingCache<K, V>
+        implements NonLoadableCache<K, V>
```

Comment:
What's the rationale to have `NonLoadableCache` interface and `NonLoadableCacheImpl` as separate classes?
I.e. would anything bad happen if the implementation class was called `NonLoadableCache`, was public and had a package-private constructor?